classes/Cart.php

```diff
@@ -2113,13 +2113,21 @@ public function getOrderTotal(
 
         if ($type == Cart::ONLY_PHYSICAL_PRODUCTS_WITHOUT_SHIPPING) {
             foreach ($products as $key => $product) {
-                if ($product['is_virtual']) {
+                if (!empty($product['is_virtual'])) {
                     unset($products[$key]);
                 }
             }
             $type = Cart::ONLY_PRODUCTS;
         }
 
+        if ($type == Cart::ONLY_PRODUCTS) {
```

Comment:
ha, it's interesting that it's working for you, sometimes we need to go through `CartRules` and check if `reduction_product` is something from cart products, but yeah we all know how it's going with cart rules 😅

Reply:
The whole Cart & Cart rules have to be refactored :sweat: